src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -242,7 +243,12 @@ void CalorimeterHitReco::process(
         const auto pos = local.nominal().worldToLocal(gpos);
         std::vector<double> cdim;
         // get segmentation dimensions
-        auto segmentation_type = m_converter->findReadout(local).segmentation().type();
+        auto segmentation = m_converter->findReadout(local).segmentation();
+        auto segmentation_type = segmentation.type();
+        while (segmentation_type == "MultiSegmentation"){
+            segmentation = ((MutltiSegmentation&)segmentation).subsegmentation(cellID);
```

Comment:
Avoid c-style casts.

Reply:
I am not sure how to do this.  Please enlighten me with a suggested change.

Reply:
I think you should be able to use a dynamic_cast on reference types here.

Reply:
Something like
```
        Segmentation& segmentation = m_converter->findReadout(local).segmentation();
        auto segmentation_type = segmentation.type();
        while (segmentation_type == "MultiSegmentation"){
            auto& multi_segmentation = dynamic_cast<Segmentation&>(segmentation);
            segmentation= multi_segmentation.subsegmentation(cellID);
```
but I'm entirely unable to test this here right now.

Reply:
Probably works with first auto instead of explicit Segmentation&.